site/content/docs/tutorials/supervisor-without-concierge-demo.md

```diff
@@ -0,0 +1,243 @@
+clusters all with a single Pinniped Supervisor.
+
+## Prerequisites
+
+1. A Kubernetes cluster of a type which allows you to adjust the command-line flags of `kube-apiserver`.
+
+   Don't have a cluster handy? Consider using [kind](https://kind.sigs.k8s.io/) on your local machine.
+   See below for an example of using kind.
+
+1. A kubeconfig where the current context points to the cluster and has administrator-like
+   privileges on that cluster
+
+1. A Pinniped Supervisor already installed and running on another cluster, and already configured with
+   a working FederationDomain, TLS certificates, and an external identity provider
+   (e.g. an OIDCIdentityProvider, LDAPIdentityProvider, or ActiveDirectoryIdentityProvider).
+
+   Don't have a Supervisor ready? Please refer to the other documents on this site to help you get one up and running
+   and sufficiently configured.
+   This tutorial does not show to install and configure the Pinniped Supervisor. Those steps are
+   shown in the [Concierge with Supervisor tutorial]({{< ref "concierge-and-supervisor-demo" >}}),
+   so if you would like you could follow the steps of that tutorial to install and configure a Supervisor
+   before returning to this tutorial.
+
+   One Pinniped Supervisor can provide authentication for many Kubernetes clusters.
+
+## How to configure the kube-apiserver flags
+
+The `kube-apiserver` command-line flags may be configured on each cluster to trust your Pinniped Supervisor to provide
+user authentication to that cluster. Add the following flags to the existing list of flags for your cluster:
+
+```bash
+# Make this exactly match the spec.issuer of your Supervisor's FederationDomain.
+# Note: It seems like the kube-apiserver pod cannot resolve `cluster.local`
+# DNS names, so don't use one of those DNS names as the issuer.
+--oidc-issuer-url="https://my-supervisor.example.com/my-issuer"
+
+# This is only required if the kube-apiserver pod is not going to trust your
+# Supervisor's FederationDomain's TLS certificates, e.g. if you used a
+# self-signed CA. Make this match where you mounted the CA PEM file into
+# your control plane node's filesystem, which must be under a directory
+# that the kube-apiserver container is going to volume mount.
+--oidc-ca-file="/etc/ca-certificates/supervisor/root-ca.pem"
+
+# Choose a unique value for each cluster here. By making this unique, the
+# Supervisor will be able to issue ID tokens for this cluster that cannot
+# be used on any other cluster, which improves security. Do not use the
+# special value "pinniped-cli" or any value that contains the substring
+# ".pinniped.dev", because these special values are reserved for other
+# purposes.
+--oidc-client-id="my-cluster-342klb7h"
+
+# Use these exact values. These are based on how the Supervisor issues ID
+# tokens. Do not change these values.
+--oidc-required-claim="azp=pinniped-cli"
```

Comment:
Note to self and reviewers: should we recommend the use of `--oidc-required-claim="azp=pinniped-cli"` or not?

Reply:
If it is not included, what might happen?

Reply:
Now that I think about it, maybe I should remove this.

When the Supervisor issues an ID token during an initial login, it will have the `aud` (audience) claim set to the name of the client who requested it. This would be `pinniped-cli` for the kubectl use case, or could be the name of a registered `OIDCClient` for a web app use case (which always starts with `client.oauth.pinniped.dev-`).

Usually the next step of a login flow (although it is configurable) is for the CLI or web app to go back to the Supervisor to request a new cluster-scoped ID token. This is the token that the client will send to the cluster for authentication. This cluster-scoped ID token's `aud` will be the audience specific to that cluster. The original `aud` value (which was the client name) will be moved to the `azp` field on this ID token.

However, the creator of the cluster would most likely want both kinds of clients to be able to authenticate into the cluster. `--oidc-required-claim="azp=pinniped-cli"` is like saying that the original login must have come from the `pinniped-cli` client, and not from an `OIDCClient`. But this should not be necessary. The `OIDCClient` spec already has a way for the admin to decide if that client should be allowed to authenticate to clusters or not.

What's more important is that the cluster's `--oidc-client-id="my-cluster-342klb7h"` value should not be `pinniped-cli` or include `.pinniped.dev`. I had already mentioned that in the doc, but perhaps I should explain more about why that is important in the doc. Avoiding those values makes it impossible for the original ID token issued by the Supervisor on an initial login to be used against a cluster. This means that the client must perform the extra request to get the cluster-scoped ID token, which allows other controls to be put in place on that request (e.g. disabling for specific `OIDCClients`), and also prevents the use of that token across multiple clusters (to avoid replay attacks).

Reply:
Force pushed the branch to update this and some other details in the doc.